solc/CommandLineInterface.cpp

```diff
@@ -225,6 +221,28 @@ static bool needsHumanTargetedStdout(po::variables_map const& _args)
 	return false;
 }
 
+static bool needsHumanTargetedStdout(po::variables_map const& _args)
+{
+	if (_args.count(g_argGas))
+		return true;
+	if (_args.count(g_argOutputDir))
+		return false;
+	if(needsStdout(_args))
+		return true;
+	return false;
+}
+
+static bool checkOutputComponent(po::variables_map const& _args)
```

Comment:
What exactly does this function do? Does it return true if and only if it creates a file?

Reply:
`needsHumanTargetedStdout` returns `true` if it outputs to stdout and `false` if it creates a file.
Sample outputs to clarify my point:
./solidity/build/solc/solc -o ./ --userdoc SimpleContract.sol 
Warning: This is a pre-release compiler version, please do not use it in production.
Requested files created.

./solidity/build/solc/solc --userdoc SimpleContract.sol 
Warning: This is a pre-release compiler version, please do not use it in production.

======= SimpleContract.sol:SimpleContract =======
User Documentation
{
  "methods" : {}
}

======= SimpleContract.sol:SimpleContract1 =======
User Documentation
{
  "methods" : {}
}

======= temp.sol:temp =======
User Documentation
{
  "methods" : {}
}

======= temp1.sol:temp1 =======
User Documentation
{
  "methods" : {}
}

Reply:
My point was: Please either add a comment to the code that explains the pupose of the function in a concise way or rename the function so that it gets clear from its name. ("check" can mean anything).